### PR TITLE
Add readmes for performance tests [11763]

### PR DESCRIPTION
**Pull request by richiware**

### CODE REVIEW

File: test/performance/throughput/README.md [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+The publication node, besides publishing fixed-size data in a custom rate, is in charge of controlling the synchronization
+with the subscription nodes and gathering the results. Both roles are provided by the same [utility](#usage) which can be
+[compiled](#compilation) with Fast-DDS.
+
+Also this directory provides a [python script](#python-launcher) which helps launching both nodes in the same system.
+
+## Throughput measure
+
+The test consists on sending during a period of time as much as possible bursts of samples separated by a fix *recovery*
+time. While the test doesn't exceed the test time, the publication node will try in a loop to send a burst of nth samples and
```
Comment (rsanchez15) [inline test/performance/throughput/README.md:15]:
```suggestion
time. While the test does not exceed the test time, the publication node will try in a loop to send a burst of `n` samples and
```

File: test/performance/throughput/README.md [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+sleep for the *recovery* time. After the test exceeds the test time, the publication node will gather all information
+from the subscription nodes to show you at the end.
+
+Our utility is able to execute at the same time several tests with different setups.
```
Comment (rsanchez15) [inline test/performance/throughput/README.md:19]:
```suggestion
The throughput test application is able to execute at the same time several tests with different setups.
```

File: test/performance/throughput/README.md [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+These are the elements you can configure:
```
Comment (rsanchez15) [inline test/performance/throughput/README.md:20]:
```suggestion
The elements that can be configured are the following:
```

File: test/performance/throughput/README.md [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+
+- Size of the samples.
+- Number of samples for each burst.
+- Recovery time -- time of each break between sending a burst of samples and the next one.
+- Test time -- maximum time the throughput test must be running.
+
+At the end of the testing the utility will show a table with the test results.
```
Comment (rsanchez15) [inline test/performance/throughput/README.md:27]:
```suggestion
At the end of the execution the utility will show a table with the test results. Below is an example of these test results.
```

File: test/performance/throughput/README.md [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+
+```
+Payload [Bytes],   Demand [sample/burst],   Recovery time [ms],    Sent [samples],   Publication time [us],   Publication sample rate [Sample/s],   Publication throughput [Mb/s],   Received [samples],   Lost [samples],   Subscription time [us],   Subscription sample rate [Sample/s],   Subscription throughput [Mb/s]
+16,                30000,                   0,                     1830000,          1002619.284,             1825219.232,                          233.628,                         1830000,              0,                1002608.786,              1825238.344,                           233.631
+32,                30000,                   0,                     1860000,          1015109.505,             1832314.634,                          469.073,                         1860000,              0,                1015109.223,              1832315.143,                           469.073
+```
+
+Each line of the table is an execution with a specific setup.
+Columns shows next information:
```
Comment (rsanchez15) [inline test/performance/throughput/README.md:36]:
```suggestion
The columns show the next information:
```

File: test/performance/throughput/README.md [hunk 6]
```diff
@@ -0,0 +1,185 @@
...
+* Received [samples] -- Number of samples received by the subscription node.
+* Lost [samples] -- Number of samples lost during the testing.
+* Subscription time [us] -- Total time the subscription was running the test.
+* Subscription sample rate [Sample/s] -- Samples rate in the subscription side.
+* Subscription throughput [Mb/s] -- Throughput measure in the subscription side.
+
+
+## Compilation
+
+You can enable the compilation of the utility using the CMake option `PERFORMANCE_TESTS`.
```
Comment (rsanchez15) [inline test/performance/throughput/README.md:54]:
```suggestion
This utility can be enabled by using the CMake option `PERFORMANCE_TESTS`.
```

File: test/performance/throughput/README.md [hunk 7]
```diff
@@ -0,0 +1,185 @@
...
+Following our [*Installation from sources* guide](https://fast-dds.docs.eprosima.com/en/latest/installation/sources/sources_linux.html)
+ you may compile Fast-DDS and this utility using next command.
```
Comment (rsanchez15) [inline test/performance/throughput/README.md:56]:
```suggestion
Following the Fast DDS [*Installation from sources* guide](https://fast-dds.docs.eprosima.com/en/latest/installation/sources/sources_linux.html),
 Fast-DDS and this utility can be compiled executing the next command.
```

File: test/performance/throughput/README.md [hunk 8]
```diff
@@ -0,0 +1,185 @@
...
+
+```
+colcon build --cmake-args -DPERFORMANCE_TESTS=ON
+```
+
+You can find this utility executable in the building directory.
```
Comment (rsanchez15) [inline test/performance/throughput/README.md:62]:
```suggestion
The throughput test executable can be found in the `build` directory.
```

File: test/performance/throughput/README.md [hunk 9]
```diff
@@ -0,0 +1,185 @@
...
+├── CTestTestfile.cmake
+├── Makefile
+├── ThroughputTest  <=== Throughput test utility
+└── xml
+```
+
+## Usage
+
+The utility is able to have the publication role or the subscription role.
+Also it is able to have both roles, useful for testing special Fast-DDS mechanisms such as *intraprocess communication*
```
Comment (rsanchez15) [inline test/performance/throughput/README.md:77]:
```suggestion
Also it is able to have both roles, useful for testing special Fast DDS mechanisms such as *intraprocess communication*
```

File: test/performance/throughput/README.md [hunk 10]
```diff
@@ -0,0 +1,185 @@
...
+This example will execute two tests, one with a recovery time of 5 milliseconds and the other one with a recovery time of 10 milliseconds.
+
+The file with the demands has the format of each line the data size separated of all demands with a `;` character.
+```
+16;100;1000
+32;100;1000
+```
+
+This examples will execute four tests: one sending bursts of 100 samples of 16 bytes, other test sending bursts of 1000 samples of 16 bytes,
+, other seding bursts of 100 samples of 32 bytes and the last one sending bursts of 1000 samples of 32 bytes.
```
Comment (rsanchez15) [inline test/performance/throughput/README.md:139]:
```suggestion
, and other seding bursts of 100 samples of 32 bytes and the last one sending bursts of 1000 samples of 32 bytes.
```

File: test/performance/throughput/README.md [hunk 11]
```diff
@@ -0,0 +1,185 @@
...
+# Publication node
+$ ThroughtputTest publisher --reliability=beteffort --domain 0 --shared_memory=off --time=10 --recovery_time=90 --demand=100 --msg_size=2097152
+
+# Subscription node
+$ ThroughtputTest subscriber --reliability=beteffort --domain 0 --shared_memory=off
+```
+
+## Python launcher
+
+The directory also comes with a python script which automates the execution of the test nodes.
```
Comment (rsanchez15) [inline test/performance/throughput/README.md:163]:
```suggestion
The directory also comes with a Python script which automates the execution of the test nodes.
```

**Review by rsanchez15 — CHANGES_REQUESTED** [review]
Throughput test review

File: test/performance/throughput/README.md [hunk 12]
```diff
@@ -0,0 +1,185 @@
+# Throughput testing
+
+This directory provides all you need for [measuring the network throughput](#throughput-measure) of Fast-DDS in your environment.
+
+Throughput measure implies having at least two nodes: one with the publication role and the other one with the subscription role.
+The publication node, besides publishing fixed-size data in a custom rate, is in charge of controlling the synchronization
+with the subscription nodes and gathering the results. Both roles are provided by the same [utility](#usage) which can be
+[compiled](#compilation) with Fast-DDS.
```
[the remark this answers is not in the document]
Comment (richiware) [reply test/performance/throughput/README.md]: Done

File: test/performance/throughput/README.md (hunk 1 above)
Comment (richiware) [reply test/performance/throughput/README.md]: Done

**Review by rsanchez15 — APPROVED** [review]

**Comment by richiware** [comment]
Failed tests not related.

**Pull request merged**